### ISSUE TITLE
shellcheck not detecting bad substitution in dash and bash when using ${#$var}

### ISSUE THREAD

**Issue by nihilus** [koalaman/shellcheck#847]
#### For new checks and feature suggestions
- [x ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh
var="foo"
echo "${#$var}"

```

#### Here's what shellcheck currently says:
No issues detected!

#### Here's what I wanted or expected to see:
Something about 'bad substitution'.

**Comment by mmlb** [comment]
@nihilus just fyi that a [tool](https://github.com/mvdan/sh) I use along with shellcheck catches this, might want to add to the  belt.

```sh
cat <<'EOF' | shfmt                                                                                                     
var="foo"
echo "${#$var}"
EOF
2:11: $ cannot be followed by a word 
```

**Comment by nihilus** [comment]
@mmlb nice tool, I integrated it today as part of our buildprocess.
